assets/Mods/BPML_GenericFunctions/Scripts/main.lua

```diff
@@ -65,3 +65,148 @@ RegisterCustomEvent("ConstructPersistentObject", function(ParamContext, ParamCla
+--- NOTE: Don't forget to unregister your hooks whenever your Actor gets destroyed.
+RegisterCustomEvent("RegisterHookFromBP", function (Context, HookName, FunctionToCall)
+    -- Type checking similar to PrintToModLoader
+    if HookName:get():type() ~= "FString" then error(string.format("RegisterHookFromBP Param #1 must be FString but was %s", HookName:get():type())) end
+    if FunctionToCall:get():type() ~= "FString" then error(string.format("RegisterHookFromBP Param #2 must be FString but was %s", FunctionToCall:get():type())) end
+
+    local HookName_AsString = HookName:get():ToString()
+
+    -- Safeguard for checking if the function hook target is actually valid
+    local OriginalFunction = StaticFindObject(HookName_AsString)
+    if OriginalFunction == nil or not OriginalFunction:IsValid() or OriginalFunction:type() ~= "UFunction" then
+        error(string.format("Tried to hook invalid function '%s'", HookName_AsString))
+    end
+
+    local FunctionName = FunctionToCall:get():ToString()
+    local CallbackFunction = Context:get()[FunctionName]
+    if CallbackFunction == nil or not CallbackFunction:IsValid() or CallbackFunction:type() ~= "UFunction" then
+        error(string.format("Function '%s' doesn't exist in '%s'", FunctionName, Context:get():GetFullName()))
+    end
+
+    if RegisteredBPHooks[HookName_AsString] == nil then
+        RegisteredBPHooks[HookName_AsString] = {}
+    end
+
+    local ExpectedContextClass = OriginalFunction:GetOuter()
+
+    if not ExpectedContextClass:IsValid() or not ExpectedContextClass:IsClass() then
+        error("Context Class was invalid or not a Class")
+    end
+
+    local OriginalTypes = {}
+    local CallbackTypes = {}
+
+    OriginalFunction:ForEachProperty(function(Property)
+        if Property:HasAnyPropertyFlags(
+            EPropertyFlags.CPF_ConstParm | 
+            EPropertyFlags.CPF_Parm | 
+            EPropertyFlags.CPF_OutParm | 
+            EPropertyFlags.CPF_ReturnParm
+        ) then
+            table.insert(OriginalTypes, Property)
+        end
+    end)
+
+    CallbackFunction:ForEachProperty(function(Property)
+        if Property:HasAnyPropertyFlags(
+            EPropertyFlags.CPF_ConstParm | 
+            EPropertyFlags.CPF_Parm | 
+            EPropertyFlags.CPF_OutParm | 
+            EPropertyFlags.CPF_ReturnParm
+        ) then
+            table.insert(CallbackTypes, Property)
+        end
+    end)
+
+    if #OriginalTypes+1 ~= #CallbackTypes then
+        error("Param count did not match!")
+    end
+
+    if CallbackTypes[1]:GetPropertyClass():GetFullName() ~= ExpectedContextClass:GetFullName() then
+        error(string.format("Context Class did not match the expected Class '%s'", ExpectedContextClass:GetFName():ToString()))
+    end
+
+    for i = 1, #OriginalTypes, 1 do
+        -- Using i+1 here to skip Context from our Callback function, maybe there's a cleaner/better way?
```

Comment:
Is it not possible to do `for i = 2` instead of doing +1 later ?
You'd have to do -1 for the error logging but it's still cleaner that way.

Reply:
I initially did the +1 because the tables are naturally mismatched due to the original function not including Context as a param property. I don't know what the cleanest way to go about it would be, but I could just add the `ExpectedContextClass` as the first table entry in `OriginalTypes` or I could add some filler value in.

Reply:
> I initially did the +1 because the tables are naturally mismatched due to the original function not including Context as a param property. I don't know what the cleanest way to go about it would be, but I could just add the `ExpectedContextClass` as the first table entry in `OriginalTypes` or I could add some filler value in.

Not sure why `CallbackTypes` would contain context as a param but not `OriginalTypes`, maybe I'm having a brain fart here.

Some filler makes more sense to me than adding context because we don't need the context there and we can use something trivial instead and on top of that it makes more sense to anyone reading the code.

It probably doesn't really matter that much to be honest if it's left like it currently is, I just wanted to address a possible low hanging fruit change but it appears I was wrong about how low it was actually hanging.

Reply:
I came up with a potentially better solution where I just pop out Context from `CallbackTypes` after we're done using it.